internal/cmd/templatefuncs.go

```diff
@@ -190,23 +190,23 @@ func (c *Config) fromJsonTemplateFunc(s string) any {
 // fromJsoncTemplateFunc parses s as JSONC and returns the result. In contrast
 // to encoding/json, numbers are represented as int64s or float64s if possible.
 func (c *Config) fromJsoncTemplateFunc(s string) any {
-	var value map[string]any
+	var value any
 	if err := chezmoi.FormatJSONC.Unmarshal([]byte(s), &value); err != nil {
 		panic(err)
 	}
 	return value
 }
 
 func (c *Config) fromTomlTemplateFunc(s string) any {
-	var value map[string]any
+	var value any
```

Comment:
I don't think that TOML allows anything but a dictionary at the root, so this is probably fine as it was.

Reply:
Thanks! Fixed in https://github.com/twpayne/chezmoi/commit/e68328eecd27255aac0a44ca33055f9c02813238.